crypto/evp/pkey_mac.c

```diff
@@ -53,19 +53,25 @@ static int pkey_mac_init(EVP_PKEY_CTX *ctx)
     int nid = ctx->pmeth->pkey_id;
     EVP_MAC *mac = EVP_MAC_fetch(ctx->libctx, OBJ_nid2sn(nid), ctx->propquery);
 
-    if (mac == NULL) {
-        EVPerr(EVP_F_PKEY_MAC_INIT, EVP_R_FETCH_FAILED);
-        return 0;
-    }
+    /*
+     * mac == NULL may actually be ok in some situations. In an
+     * EVP_PKEY_new_mac_key() call a temporary EVP_PKEY_CTX is created with
+     * default libctx. We don't actually need the underlying MAC to be present
```

Comment:
Are you saying the default libctx is the fips provider for example (otherwise this makes not much sense to me).

Reply:
In the test case that failed the default libctx had the legacy provider loaded (only), but there was an alternative libctx that needed to use the EVP_PKEY being created. That libctx had the FIPS provider loaded. So HMAC did not exist in the default libctx and EVP_PKEY_new_mac_key() failed - even though I was eventually going to use the EVP_PKEY in a libctx that did have HMAC.